src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -152,41 +118,30 @@ public static <B> DeferredRegister<B> createOptional(ResourceLocation registryNa
     private SetMultimap<TagKey<T>, Supplier<T>> optionalTags;
     private boolean seenRegisterEvent = false;
 
-    private DeferredRegister(ResourceKey<? extends Registry<T>> registryKey, String modid, boolean optionalRegistry)
+    private DeferredRegister(ResourceKey<? extends Registry<T>> registryKey, String modid)
     {
-        this.registryKey = registryKey;
-        this.modid = modid;
-        this.optionalRegistry = optionalRegistry;
-    }
-
-    private DeferredRegister(IForgeRegistry<T> reg, String modid)
-    {
-        this(reg.getRegistryKey(), modid, false);
+        this.registryKey = Objects.requireNonNull(registryKey);
+        this.modid = Objects.requireNonNull(modid);
     }
 
     /**
-     * Adds a new supplier to the list of entries to be registered, and returns a RegistryObject that will be populated with the created entry automatically.
-     *
-     * @param name The new entry's name, it will automatically have the modid prefixed.
+     * Adds a new supplier to the list of entries to be registered, and returns a DeferredHolder pointing to the to-be-registered object.
+     * <p>
+     * @param name The new entry's name. The modid from the constructor will be used as the namespace.
```

Comment:
This is a confusing comment in my eyes, it should specify that it uses the namespace that was declared when the DeferredRegister was created. Right now, it is too generic. (Also, I potentially think we should make the `modid` field generic and rename it to just `namespace`.)